<launch>
    <!-- 相机的IP地址、端口、用户名、密码和通道参数 -->
    <arg name="ip" default="192.168.1.160"/>
    <arg name="port" default="8000"/>
    <arg name="username" default="admin"/>
    <arg name="password" default="Ydkj147258"/>
    <arg name="channel" default="1"/>
    <arg name="combined_topic" default="combined_image"/>
    <arg name="left_topic" default="left_image"/>
    <arg name="right_topic" default="right_image"/>

    <!-- 启动ROS节点 -->
    <node pkg="hikrobot_camera" type="hkvision" name="hikrobot_camera" output="screen">
        <!-- 传递参数给节点 -->
        <param name="ip" value="$(arg ip)"/>
        <param name="port" value="$(arg port)"/>
        <param name="username" value="$(arg username)"/>
        <param name="password" value="$(arg password)"/>
        <param name="channel" value="$(arg channel)"/>
        <param name="topic_name" value="$(arg combined_topic)"/>
        <param name="left_topic_name" value="$(arg left_topic)"/>
        <param name="right_topic_name" value="$(arg right_topic)"/>
    </node>
</launch>
